src/patterns/scheduled-ecs-task.ts

```diff
@@ -0,0 +1,239 @@
+
+  if (ecrContainer) {
+    return ecrContainer;
+  } else {
+    return ContainerImage.fromRegistry(config.id ?? "ubuntu:focal");
+  }
+};
+
+/**
+ * Pattern which creates all of the resources needed to invoke a fargate task on a schedule.
+ *
+ * The task will be wrapped in a step function to allow for easier triggering and alarming on failure.
+ *
+ * For all configuration options, see [[`GuScheduledEcsTaskProps`]].
+ *
+ * Note that if your task reliably completes in less than 15 minutes then you should probably use a [[`GuScheduledLambda`]] instead. This
+ * pattern was mainly created to work around the 15 minute lambda timeout.
+ */
+export class GuScheduledEcsTask {
+  constructor(scope: GuStack, props: GuScheduledEcsTaskProps) {
+    const distBucket = GuDistributionBucketParameter.getInstance(scope).valueAsString;
+
+    const timeout = props.taskTimeoutInMinutes ?? 15;
+
+    const cpu = props.cpu ?? 2048;
+    const memory = props.memory ?? 4096;
+
+    const cluster = new Cluster(scope, `${props.app}-cluster`, {
+      clusterName: `${props.app}-cluster-${props.stage}`,
+      enableFargateCapacityProviders: true,
+      vpc: props.vpc,
+    });
+
+    const taskDefinition = new TaskDefinition(scope, `${props.app}-task-definition`, {
+      compatibility: Compatibility.FARGATE,
+      cpu: cpu.toString(),
+      memoryMiB: memory.toString(),
+    });
+
+    taskDefinition.addContainer("scheduled-task-container", {
+      image: getContainer(props.containerConfiguration),
+      entryPoint: props.taskCommand ? ["/bin/sh"] : undefined,
+      command: props.taskCommand ? ["-c", `${props.taskCommand}`] : undefined,
+      cpu,
+      memoryLimitMiB: props.memory,
+      logging: LogDrivers.awsLogs({
+        streamPrefix: props.app,
+        logRetention: 14,
+      }),
+    });
+
+    const fetchArtifact = new PolicyStatement();
+    fetchArtifact.addActions("s3:GetObject");
+    fetchArtifact.addResources(`arn:aws:s3:::${distBucket}/*`);
+    taskDefinition.addToTaskRolePolicy(fetchArtifact);
+    (props.customTaskPolicies ?? []).forEach((p) => taskDefinition.addToTaskRolePolicy(p));
+
+    const task = new EcsRunTask(scope, `${props.app}-task`, {
+      cluster: cluster,
+      launchTarget: new EcsFargateLaunchTarget({
+        platformVersion: FargatePlatformVersion.LATEST,
+      }),
+      taskDefinition: taskDefinition,
+      integrationPattern: IntegrationPattern.RUN_JOB,
+      resultPath: "DISCARD",
+      timeout: Duration.minutes(timeout),
+      securityGroups: props.securityGroups ?? [],
+    });
+
+    const stateMachine = new StateMachine(scope, `${props.app}-state-machine`, {
+      definition: task,
+      stateMachineName: `${props.app}-${props.stage}`,
+    });
+
+    new Rule(scope, `${props.app}-schedule-rule`, {
+      schedule: props.schedule,
+      targets: [new SfnStateMachine(stateMachine)],
+    });
+
+    if (props.alarmSnsTopicArn) {
```

Comment:
Other [constructs](https://github.com/guardian/cdk/blob/55c689e584b08e600772eea74f624112f34ec7f7/src/patterns/api-lambda.ts#L35) and patterns follow a slightly different approach here; alarms are always on unless `noMonitoring: true`. Could we take a similar approach here?

Reply:
👍🏻 addressed in https://github.com/guardian/cdk/pull/791/commits/01099d273e7a80db9c294f2ccb1b28c0fbb569fd